packages/nc-gui-v2/components/smartsheet/sidebar/menu/ApiSnippet.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+import HTTPSnippet from 'httpsnippet'
+import { useClipboard } from '@vueuse/core'
+import { useToast } from 'vue-toastification'
+import { ActiveViewInj, MetaInj } from '~/context'
+
+const props = defineProps<Props>()
+
+const emits = defineEmits(['update:modelValue'])
+
+const toast = useToast()
+
+interface Props {
+  modelValue: boolean
+}
+
+const { project } = $(useProject())
+const { appInfo } = $(useGlobal())
+const meta = $(inject(MetaInj))
+const view = $(inject(ActiveViewInj))
+const { xWhere } = useSmartsheetStoreOrThrow()
+const { queryParams } = $(useViewData(meta, view as any, xWhere))
+const { copy } = useClipboard()
+const { $state } = useNuxtApp()
+
+let vModel = $(useVModel(props, 'modelValue', emits))
+
+const langs = [
+  {
+    name: 'shell',
+    clients: ['curl', 'wget'],
+  },
+  {
+    name: 'javascript',
+    clients: ['axios', 'fetch', 'jquery', 'xhr'],
+  },
+  {
+    name: 'node',
+    clients: ['axios', 'fetch', 'request', 'native', 'unirest'],
+  },
+  {
+    name: 'nocodb-sdk',
+    clients: ['javascript', 'node'],
+  },
+  {
+    name: 'php',
+  },
+  {
+    name: 'python',
+    clients: ['python3', 'requests'],
+  },
+  {
+    name: 'ruby',
+  },
+  {
+    name: 'java',
+  },
+  {
+    name: 'c',
+  },
+]
+
+const selectedClient = $ref<string | undefined>(langs[0].clients && langs[0].clients[0])
```

Comment:
Seems a bit confusing, why confirm the existence of `langs[0].clients` if it is definetly defined anyway?
TypeScript should know that it has to be defined

Reply:
@bcakmakoglu Because clients can be undefined as well. It is optional. We hide the dropdown for client in such situations.